tiledb/sm/c_api/tiledb.cc

```diff
@@ -4956,7 +4957,25 @@ int32_t tiledb_fragment_info_get_fragment_name(
       sanity_check(ctx, fragment_info) == TILEDB_ERR)
     return TILEDB_ERR;
 
-  throw_if_not_ok(fragment_info->fragment_info_->get_fragment_name(fid, name));
+  *name = (new std::string(fragment_info->fragment_info_->fragment_name(fid)))
```

Comment:
Calling `new` here is a resource leak. I understand that it's on purpose, but it definitely needs two more things:
* A comment stating what's happening.
* A log message warning about the resource leak.